sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionManager.java

```diff
@@ -224,16 +224,21 @@ private AmqpErrorContext getErrorContext() {
     }
 
     /**
-     * Creates an unnamed session receive link.
+     * Creates an session receive link.
      *
-     * @return A Mono that completes with an unnamed session receive link.
+     * @return A Mono that completes with an session receive link.
      */
     private Mono<ServiceBusReceiveLink> createSessionReceiveLink() {
-        final String linkName = StringUtil.getRandomString("session-");
+        final String sessionId = receiverOptions.getSessionId();
 
+        final String linkName = (sessionId != null)
+            ? sessionId
```

Comment:
We should still append a random string to the name of the session to get a unique link name.

Reply:
If  we make linkName random for same SessionId, It will try to create a new Link for same session (Example SessionId= GREEN). And if  a Link already exists for GREEN, we will not be able to get the lock.